src/node/eviction.cpp

```diff
@@ -175,14 +176,22 @@ void ProtectEvictionCandidatesByRatio(std::vector<NodeEvictionCandidate>& evicti
     EraseLastKElements(eviction_candidates, ReverseCompareNodeTimeConnected, remaining_to_protect);
 }
 
-[[nodiscard]] std::optional<NodeId> SelectNodeToEvict(std::vector<NodeEvictionCandidate>&& vEvictionCandidates)
+[[nodiscard]] std::optional<NodeId> SelectNodeToEvict(std::vector<NodeEvictionCandidate>&& vEvictionCandidates, bool force)
 {
     // Protect connections with certain characteristics
 
     ProtectNoBanConnections(vEvictionCandidates);
 
     ProtectOutboundConnections(vEvictionCandidates);
 
+    // Hang on to one random node to evict if forced
+    std::optional<NodeId> force_evict;
+    if (vEvictionCandidates.size() > 0) {
```

Comment:
This block can be skipped if `!force`

```suggestion
    if (vEvictionCandidates.size() > 0 && force) {
```